api/src/services/files.ts

```diff
@@ -86,6 +86,11 @@ export class FilesService extends ItemsService<File> {
 			payload.filename_disk = primaryKey + (fileExtension || '');
 		}
 
+			// If the filename_disk doesn't include an extension, append it
+			if (!path.extname(payload.filename_disk!)) {
+				payload.filename_disk = payload.filename_disk + (fileExtension || '');
```

Comment:
```suggestion
				payload.filename_disk += (fileExtension || '');
```

Might be a better option?